src/fields/Fields.cpp

```diff
@@ -874,19 +844,19 @@ Fields::SolvePoissonPsiExmByEypBxEzBz (amrex::Vector<amrex::Geometry> const& geo
 
     PhysConst phys_const = get_phys_const();
 
+    EnforcePeriodic(true, {Comps[WhichSlice::This]["jx"],
+                           Comps[WhichSlice::This]["jy"],
+                           Comps[WhichSlice::This]["rhomjz"]});
     for (int lev=0; lev<current_N_level; ++lev) {
-        if (!m_extended_solve && lev==0) {
-            FillBoundary(geom[lev].periodicity(), lev, WhichSlice::This, "jx", "jy", "rhomjz");
-        }
         // interpolate rhomjz to lev from lev-1 in the domain edges
         LevelUpBoundary(geom, lev, WhichSlice::This, "rhomjz",
-            m_poisson_nguards, -m_slices_nguards);
+            amrex::IntVect{0, 0, 0}, -m_slices_nguards + amrex::IntVect{1, 1, 0});
```

Comment:
m_poisson_nguards was only non zero for extended_solve so I replaced it with zero.

Reply:
I subtract one from the slice guards for interpolation is this should be all that is necessary, since rhomjz comes from the normal current deposition with the smaller shape factor compared to the explicit deposition. I found that this helps with stability when using MR